yaaredis/pool.py

```diff
@@ -647,3 +648,235 @@ def get_node_by_slot(self, slot):
+                    'host': connection_kwargs.pop('host'),
+                    'port': str(connection_kwargs.pop('port')),
+                }]
+
+        self.max_connections = max_connections or 2 ** 31
+        self.nodes = NodeManager(
+            startup_nodes,
+            reinitialize_steps=reinitialize_steps,
+            skip_full_coverage_check=skip_full_coverage_check,
+            max_connections=self.max_connections,
+            nodemanager_follow_cluster=nodemanager_follow_cluster,
+            **connection_kwargs,
+        )
+        self.initialized = False
+
+        self.connections = {}
+        self.connection_kwargs = connection_kwargs
+        self.connection_kwargs['readonly'] = readonly
+        self.readonly = readonly
+        self.max_idle_time = max_idle_time
+        self.idle_check_interval = idle_check_interval
+        self.reset()
+
+        if 'stream_timeout' not in self.connection_kwargs:
+            self.connection_kwargs['stream_timeout'] = ClusterConnectionPool.RedisClusterDefaultTimeout
+
+    def __repr__(self):
+        """
+        Returns a string with all unique ip:port combinations that this pool
+        is connected to
+        """
+        return '{}<{}>'.format(
+            type(self).__name__,
+            ', '.join([self.connection_class.description.format(**node)
+                       for node in self.nodes.startup_nodes]),
+        )
+
+    async def initialize(self):
+        if not self.initialized:
+            await self.nodes.initialize()
+            self.initialized = True
+
+    def _make_connection_queue(self):
+        res = asyncio.LifoQueue(maxsize=self.max_connections)
+        for _ in range(self.max_connections):
+            res.put_nowait(None)
+        return res
+
+    def reset(self):
+        """Resets the connection pool back to a clean state"""
+        self.pid = os.getpid()
+        self._available_connections = defaultdict(self._make_connection_queue)
+        self._in_use_connections = defaultdict(set)
+        self._check_lock = threading.Lock()
+        self.initialized = False
+
+    def _checkpid(self):
+        if self.pid != os.getpid():
+            with self._check_lock:
+                if self.pid == os.getpid():
+                    # another thread already did the work while we waited
+                    # on the lockself.
+                    return
+                self.disconnect()
+                self.reset()
+
+    async def get_connection(self, command_name, *keys, **options):
+        # Only pubsub command/connection should be allowed here
+        if command_name != 'pubsub':
+            raise RedisClusterException(
+                "Only 'pubsub' commands can use get_connection()")
+
+        channel = options.pop('channel', None)
+
+        if not channel:
+            return await self.get_random_connection()
+
+        slot = self.nodes.keyslot(channel)
+        node = self.get_master_node_by_slot(slot)
+
+        self._checkpid()
```

Comment:
This check is unnecessary, since it happens as the first line in `get_connection_by_node()` as well.